Решил задачу с письмами

vzeten opened the pull request.

Commit:
Решил задачу с письмами

MarselYsup commented on src/main/java/school/faang/bjs2_91164_microsoft/MailSender.java:
```diff
@@ -0,0 +1,33 @@
+public class MailSender {
+    public static void main(String[] args) {
+
+        Thread thread1 = new Thread(new SenderRunnable(0, 199));
```
а если писем будет 1млн, то также будешь построчно все письма выписывать?)

Давай определим массив Thread-ов и в цикле их инциализируем, такой подход даст большую гибкость и масштабируемость 
```java

  private static final int TOTAL_MESSAGES = 1000; // Общее количество писем
    private static final int THREAD_COUNT = 5; // Количество потоков
    private static final int BATCH_SIZE = TOTAL_MESSAGES / THREAD_COUNT; // Размер блока писем для каждого потока

    public static void main(String[] args) throws InterruptedException {
        
        Thread[] threads = new Thread[THREAD_COUNT]; // Массив потоков

        // Создаем и запускаем потоки
        for (int i = 0; i < THREAD_COUNT; i++) {
// дальше в цикле определяешь задачи и вызываешь start

MarselYsup commented on src/main/java/school/faang/bjs2_91164_microsoft/SenderRunnable.java:
```diff
@@ -0,0 +1,20 @@
+
+@AllArgsConstructor
+public class SenderRunnable implements Runnable {
+    private int startIndex;
```
некритично, но для данного класса startIndex и endIndex - могут быть неизменяемыми полями - то есть после их определения, они не меняются в рамках одного объекта, поэтому можно сделать их ```final```, так ты повысишь надежность кода

Commit:
Исправил замечания
Добавил переменный
Запустил потоки через цикл
Присоединился к потокам через цикл
исправил тип переменных на final в SenderRunnable

MarselYsup commented:
вызови gradle task :checkstyleMain у тебя проблемы в код стайле, поправь их и я смержу задачу

Commit:
в классе MailSender изменил стиль переменных на camelCase
